tests/py/dynamo/lowering/test_decompositions.py

```diff
@@ -962,6 +962,414 @@ def forward(self, input):
+                {
+                    torch.ops.aten.minimum.default,
+                    torch.ops.aten.scatter.src,
+                },
+                torch.zeros(3, 5, dtype=torch.int32).cuda(),
+                "amin",
+            ),
+            (
+                "scatter_reduce_amin_one_dim_indexOne_constant",
+                1,
+                torch.tensor([[0, 1, 2, 0]]).cuda(),
+                torch.tensor([[1, 2, 3, 1]], dtype=torch.int32).cuda(),
+                {
+                    torch.ops.aten.minimum.default,
+                    torch.ops.aten.scatter.src,
+                },
+                torch.zeros(3, 5, dtype=torch.int32).cuda(),
+                "amin",
+            ),
+            (
+                "scatter_reduce_amin_one_dim_indexTwo_constant",
+                1,
+                torch.tensor([[0, 1, 2, 0], [1, 2, 1, 1]]).cuda(),
+                torch.tensor([[1, 2, 3, 1], [5, 6, 5, 5]], dtype=torch.int32).cuda(),
+                {
+                    torch.ops.aten.minimum.default,
+                    torch.ops.aten.scatter.src,
+                },
+                torch.zeros(3, 5, dtype=torch.int32).cuda(),
+                "amin",
+            ),
+            (
+                "scatter_reduce_amin_one_dim_indexTwo_constant_3D",
+                1,
+                torch.tensor(
+                    [[[0, 1, 2, 0], [1, 2, 1, 1]], [[3, 2, 1, 2], [0, 1, 2, 0]]]
+                ).cuda(),
+                torch.tensor(
+                    [[[1, 2, 3, 1], [5, 6, 5, 5]], [[2, 4, 3, 2], [1, 2, 3, 1]]],
+                    dtype=torch.int32,
+                ).cuda(),
+                {
+                    torch.ops.aten.minimum.default,
+                    torch.ops.aten.scatter.src,
+                },
+                torch.zeros(3, 5, 6, dtype=torch.int32).cuda(),
+                "amin",
+            ),
+        ]
+    )
+    def test_scatter_reduce(
+        self, _, dim, index, src, expected_ops_param, input_reduce_op, reduce_op_str
+    ):
+        class TestModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, input):
+
+                return torch.ops.aten.scatter_reduce_.two(
+                    input, dim, index, src, reduce=reduce_op_str
+                )
+
+        # Operations expected to be included in the traced graph after decompositions
+        expected_ops = expected_ops_param
+        unexpected_ops = {torch.ops.aten.scatter_reduce_.two}
+
+        input = torch.zeros(3, 5, dtype=torch.int32).cuda()
+        inputs = [input_reduce_op]
+
+        fx_graph = torch.fx.symbolic_trace(TestModule())
+        unexpected_ops_seen, expected_ops_unseen = lower_graph_testing(
+            fx_graph,
+            inputs,
+            expected_ops=expected_ops,
+            unexpected_ops=unexpected_ops,
+            min_block_size=2,
```

Comment:
Why is the min_block size = 2 here ? What  does lower_graph_testing do ? Does it use our partitioning ?

Reply:
Yes lower_graph_testing uses our partition. It is used for returning the expected ops and the unexpected ops. The default I think is 3, but sometimes the graph in the test case is too small, and the block size is lesser than 3. Thats why I set it to 1 or 2, since that is not something which we are testing explicitly in the test.
Let me try with the default 3. If it passes, I will remove the `min_block_size =2` then.